FIAT/argyris.py

```diff
@@ -4,121 +4,115 @@
+                nodes.extend(IntegralMoment(ref_el, Q_mapped, dphi * scale) for dphi in dphis[1:])
+                entity_ids[1][e].extend(range(cur, len(nodes)))
+
+            # interior dofs
+            q = degree - 6
+            if q >= 0:
+                Q = create_quadrature(ref_el, degree + q)
+                Pq = polynomial_set.ONPolynomialSet(ref_el, q, scale=1)
+                phis = Pq.tabulate(Q.get_points())[(0,) * sd]
+                scale = ref_el.volume()
+                cur = len(nodes)
+                nodes.extend(IntegralMoment(ref_el, Q, phi/scale) for phi in phis)
+                entity_ids[sd][0] = list(range(cur, len(nodes)))
+
+        elif variant == "point":
+            # edge dofs
+            for e in sorted(top[1]):
+                cur = len(nodes)
+                # normal derivatives at degree - 4 points on each edge
+                ndpts = ref_el.make_points(1, e, degree - 3)
+                nodes.extend(PointNormalDerivative(ref_el, e, pt) for pt in ndpts)
+
+                # point value at degree - 5 points on each edge
+                ptvalpts = ref_el.make_points(1, e, degree - 4)
+                nodes.extend(PointEvaluation(ref_el, pt) for pt in ptvalpts)
+                entity_ids[1][e] = list(range(cur, len(nodes)))
 
-        # edge dof
-        entity_ids[1] = {}
-        for e in sorted(top[1]):
-            # normal derivatives at degree - 4 points on each edge
-            ndpts = ref_el.make_points(1, e, degree - 3)
-            ndnds = [pnd(ref_el, e, pt) for pt in ndpts]
-            nodes.extend(ndnds)
-            entity_ids[1][e] = list(range(cur, cur + len(ndpts)))
-            cur += len(ndpts)
-
-            # point value at degree-5 points on each edge
+            # interior dofs
             if degree > 5:
-                ptvalpts = ref_el.make_points(1, e, degree - 4)
-                ptvalnds = [pe(ref_el, pt) for pt in ptvalpts]
-                nodes.extend(ptvalnds)
-                entity_ids[1][e] += list(range(cur, cur + len(ptvalpts)))
-                cur += len(ptvalpts)
-
-        # internal dof
-        entity_ids[2] = {}
-        if degree > 5:
-            internalpts = ref_el.make_points(2, 0, degree - 3)
-            internalnds = [pe(ref_el, pt) for pt in internalpts]
-            nodes.extend(internalnds)
-            entity_ids[2][0] = list(range(cur, cur + len(internalpts)))
-            cur += len(internalpts)
+                cur = len(nodes)
+                internalpts = ref_el.make_points(2, 0, degree - 3)
+                nodes.extend(PointEvaluation(ref_el, pt) for pt in internalpts)
+                entity_ids[2][0] = list(range(cur, len(nodes)))
         else:
-            entity_ids[2] = {0: []}
-
+            raise ValueError("Invalid variant for Argyris")
         super(ArgyrisDualSet, self).__init__(nodes, ref_el, entity_ids)
 
 
 class QuinticArgyrisDualSet(dual_set.DualSet):
```

Comment:
While we're mucking about with Argyris, can we just wipe out `QuinticArgyris` (dual and primal classes)?